### PR TITLE
feat: Conditional DB2 preprocessor

### PR DESCRIPTION
**Pull request by slavek-kucera**

### CODE REVIEW

File: parser_library/src/processing/preprocessors/db2_preprocessor.cpp [hunk 1]
```diff
@@ -711,12 +716,16 @@ class db2_preprocessor : public preprocessor
 
         generate_replacement(it, end, true);
 
-        return document(std::move(m_result));
+        if (m_source_translated || !m_conditional)
```
Comment (jirimosinger) [inline parser_library/src/processing/preprocessors/db2_preprocessor.cpp:719]: I guess that when `m_conditional` is `true` we can return right after call to `generate_replacement()`?

Comment (slavek-kucera) [reply parser_library/src/processing/preprocessors/db2_preprocessor.cpp:719]: ??? that's what we do here, right?

Comment (jirimosinger) [reply parser_library/src/processing/preprocessors/db2_preprocessor.cpp:719]: I meant before call to other stuff like `generate_replacement()` and `skip_process()` in order to save some processing time. Or are those functions needed to be called even in this case?

Comment (jirimosinger) [reply parser_library/src/processing/preprocessors/db2_preprocessor.cpp:719]: Ok, I think I understand now where the confusion is coming from. 

The original comment was related to `generate_replacement(document doc)` and not `generate_replacement(it, end, true);`


In other words - when we enter this function, we can immediately check if `m_conditional` is `true` and return immediately without calling `skip_process(it, end);` and `generate_replacement(it, end, true);`

Comment (slavek-kucera) [reply parser_library/src/processing/preprocessors/db2_preprocessor.cpp:719]: No we can't. We need to go through the document to verify whether there is something to replace.

**Review by jirimosinger — APPROVED** [review]

**Pull request merged**